### ISSUE TITLE
Segmentation Fault in application launch on debian derivative 

### ISSUE THREAD

**Issue by lanceofwhichwedream** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3553]
**Checklist:**

* [X] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES ASWELL**
* [X] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [X] I have double checked and can reproduce the issue

 

**OS:**
Netrunner Blackbird (Debian devirative, functionally the same as debian testing.
**GPMDP Version:**
4.6.1
**Issue Descriptions:**
GPMDP will not start. Attempting to launch the application from the start menu produces a silent failure. Attempting to launch the application via command produces this error message:

[1]    5077 segmentation fault  google-play-music-desktop-player

**Dmesg Output:**
[11621.276701] google-play-mus[5077]: segfault at a4fa30 ip 0000000000a4fa30 sp 00007fffb23ccca8 error 15 in Google Play Music Desktop Player[200000+1602000]
[11621.276707] Code: da 9d ef 9d e6 9d f2 9d f8 9d 0c 9e fa 9d 1b 9e 64 76 58 76 67 76 02 76 46 76 47 76 49 76 1e 76 3b 76 6d 76 6e 76 69 76 72 76 <c7> 7a b6 7a 60 89 33 89 5d 89 47 89 38 89 64 89 b8 76 2e 80 2c 80

**Steps to Reproduce:**
Try to launch application normally or via commandline.

**Comment by arielgrab** [comment]
same issue here in ubuntu 19.04.

error throw in command line:
Segmentation fault (core image recorded)

Dmesg:
[  954.836317] google-play-mus[21499]: segfault at a4fa30 ip 0000000000a4fa30 sp 00007ffc80d732f8 error 15 in Google Play Music Desktop Player[200000+1602000]

**Comment by rhwendt** [comment]
Checklist:

- [x] I have checked that there are no issues with similar or the same content

**YOU SHOULD CHECK CLOSED ISSUES ASWELL**

- [x] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there

- [x]  I have double checked and can reproduce the issue

OS: Ubuntu 19.04 x86_64 
Kernel: 5.0.0-13-generic 
Uptime: 14 days, 12 hours, 41 mins 
Packages: 2426 (dpkg), 10 (snap) 
Shell: zsh 5.5.1 
Resolution: 1920x1080, 1920x1080 
DE: GNOME 3.32.0 
WM: GNOME Shell 
WM Theme: Adwaita 
Theme: Yaru-dark [GTK2/3] 
Icons: Yaru [GTK2/3] 
Terminal: guake 
CPU: Intel i5-8350U (8) @ 3.600GHz 
GPU: Intel UHD Graphics 620 
Memory: 10171MiB / 15916MiB 

Steps to reproduce: 

open via command line or through gnome

Result:
```
 rwendt@rwendt-Latitude-7490  ~  google-play-music-desktop-player
[1]    27423 segmentation fault (core dumped)  google-play-music-desktop-player
```

I have attached the crash file with this

[crash.txt](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/files/3142693/_usr_share_google-play-music-desktop-player_Google.Play.Music.Desktop.Player-crash.txt)

**Comment by RandyCupic** [comment]
I had the same issue on two different machines: one is running Ubuntu 18.10 and the other one is running Ubuntu 19.04. On both machines I've installed it by downloading the .deb package from official site. Now I tried to install it manually, following this instructions: https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/master/docs/Installation_On_Debian.md and now it works fine. It seems that the .deb package is broken.

**Comment by paul** [comment]
I am experiencing the same issue using Fedora 30 with the official rpm, Fedora 29 worked fine.

OS: Fedora release 30 (Thirty)
Kernel: 5.0.10-300.fc30.x86_64

```
$ coredumpctl info
           PID: 4405 (google-play-mus)
           UID: 1000 (rando)
           GID: 1000 (rando)
        Signal: 11 (SEGV)
     Timestamp: Sat 2019-05-04 21:03:26 MDT (2min 43s ago)
  Command Line: google-play-music-desktop-player
    Executable: /usr/share/google-play-music-desktop-player/Google Play Music Desktop Player
 Control Group: /user.slice/user-1000.slice/session-2.scope
          Unit: session-2.scope
         Slice: user-1000.slice
       Session: 2
     Owner UID: 1000 (rando)
       Boot ID: e91e722441be43cf85257e290803a2b9
    Machine ID: 40aacf4b608a4b1d9479fa640e9ba3dd
      Hostname: raylan.home
       Storage: /var/lib/systemd/coredump/core.google-play-mus.1000.e91e722441be43cf85257e290803a2b9.4405.1557025406000000.lz4
       Message: Process 4405 (google-play-mus) of user 1000 dumped core.
                
                Stack trace of thread 4405:
                #0  0x0000000000a4fa30 n/a (Google Play Music Desktop Player)
```

**Comment by paul** [comment]
Update: The rpms provided in #3548 do work for me (but the 32/64 links are swapped with i386 and amd64).

**Comment by JonnoN** [comment]
> I am experiencing the same issue using Fedora 30 with the official rpm, Fedora 29 worked fine.

I am having the same issue on Fedora 29 (64 bit). It used to work, although I can't remember if it was on 28 or 29.

**Comment by magnofrey** [comment]
Same issue here in Ubuntu 19.04

**Comment by MichalKrasno** [comment]
This worked:

> Now I tried to install it manually, following [these instructions](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/master/docs/Installation_On_Debian.md), and now it works fine.

But it installed version 4.4.0, whereas current is 4.6.1 ... seems like there's a regression in the current version on Ubuntu 19.04 (at least)

**Comment by joemort** [comment]
This didn't used to be a problem for me, but recently I had to relaunch after taking some sort of updates on debian and now I get a segfault every time.

I was able to solve it by rolling back to the 4.5.0 binaries, everything newer than that segfaulted on startup.

**Comment by imaami** [comment]
## HACK MarshallOfSound to download an UP-TO-DATE package with this SIMPLE TRICK!

### DOCTORS HATE HIM!

Now that I have your attention:

I wanted to see if the latest build from master fixes this, but there's an obvious lack of download links to up-to-date packages. My findings were:

* [docs/Installation_On_Debian.md](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/master/docs/Installation_On_Debian.md) describes how to install from the release repository, but unfortunately the most recent build there is actually 4.4. Doesn't segfault, but no, just no. I want to live dangerously.

* #3548 has some links to RC builds, but those are from 2019-04-23. Relying on the author to edit a list of URLs by hand after each commit is obviously not a future-proof way of publishing links to dev builds. (Not blaming the authors here, btw, no one can reasonably demand them keep up with such manual edits IMHO. It's just not how things should work.)

* [CircleCI](https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/tree/master) actually has up-to-date package builds! But even though [this method](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3391#issuecomment-464123120) works if you're logged into CircleCI, the "Artifacts" tab which contains the download links is hidden if you're not.

So here's how you can download the latest package from CircleCI without having to log in:

Navigate to https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/tree/master and look at the topmost successful build. There's a build number on the right side of the green "success" button. As I'm writing this the number is 3366, it increments on each new build.

Select a link from the list below. None of them actually work as-is because the build number is replaced by the infinity emoji ♾️. Replace the emoji with the build number and you have a working download URL.

### Darwin

https://♾️-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip

### Debian

x86_64: https://♾️-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb
i386: https://♾️-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_i386.deb

### Redhat

x86_64: https://♾️-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm
i386: https://♾️-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.i386.rpm

**Comment by silentdis** [comment]
@imaami Thank you so much! Back up and fully working on Kubuntu 19.04 :)

**Comment by GregoryConrad** [comment]
I don't know if this helps, but the latest version available on https://www.googleplaymusicdesktopplayer.com/ worked on Ubuntu 18.04, but then when I upgraded to Ubuntu 19.04, I get a `Segmentation fault (core dumped)` when I run `google-play-music-desktop-player` from the command line.

**Comment by JeffAlyanak** [comment]
The latest version pulled from CircleCI works for me on Debian 10.

**Comment by tarka** [comment]
Confirmed, CircleCI build of 908be22 works for me on Ubuntu 19.04.

**Comment by lanceofwhichwedream** [comment]
For a while this had been fixed for me, but looks like the error came back. Still on the same debian derivative

**Comment by lanceofwhichwedream** [comment]
Moved over to Ubuntu 19.10 and this issue is still present in the version available from snap

**Comment by Archund** [comment]
Same error for me on Manjaro

```
OS: Manjaro Linux x86_64 
Host: 80SB Lenovo ideapad FLEX 4-1570 
Kernel: 4.19.85-1-MANJARO 
Uptime: 56 mins 
Packages: 1690 (pacman) 
Shell: bash 5.0.11 
Resolution: 1920x1080 
DE: Plasma 
WM: KWin 
Theme: Breath [GTK2/3] 
Icons: breeze-dark [GTK2/3] 
Terminal: guake 
CPU: Intel i7-6500U (4) @ 2.500GHz 
GPU: Intel Skylake GT2 [HD Graphics 520] 
GPU: AMD ATI Radeon R7 M260/M265 / M340/M360 / M440/M445 / 530/535 / 620/625 Mobile 
Memory: 3545MiB / 15893MiB 
```
GPMDP was installed using the PKGBUILD from [AUR](https://aur.archlinux.org/packages/gpmdp/)

Running from command line or desktop causes a segmentation fault
`journalctl | grep gpmdp` gives :
```

dic 05 23:00:58 ARC-flex audit[13646]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=2 pid=13646 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:00:58 ARC-flex kernel: gpmdp[13646]: segfault at a4fa30 ip 0000000000a4fa30 sp 00007ffd7f7f6278 error 15 in Google Play Music Desktop Player[200000+1602000]
dic 05 23:00:58 ARC-flex kernel: audit: type=1701 audit(1575608458.185:105): auid=1000 uid=1000 gid=1000 ses=2 pid=13646 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:00:58 ARC-flex systemd-coredump[13648]: Process 13646 (gpmdp) of user 1000 dumped core.
dic 05 23:01:02 ARC-flex audit[13662]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=2 pid=13662 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:01:02 ARC-flex kernel: gpmdp[13662]: segfault at a4fa30 ip 0000000000a4fa30 sp 00007ffec9163a98 error 15 in Google Play Music Desktop Player[200000+1602000]
dic 05 23:01:02 ARC-flex kernel: audit: type=1701 audit(1575608462.162:108): auid=1000 uid=1000 gid=1000 ses=2 pid=13662 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:01:02 ARC-flex systemd-coredump[13664]: Process 13662 (gpmdp) of user 1000 dumped core.
dic 05 23:06:23 ARC-flex sudo[16299]:      arc : TTY=pts/3 ; PWD=/home/arc ; USER=root ; COMMAND=/usr/bin/pacman -R gpmdp
dic 05 23:09:27 ARC-flex sudo[17329]:      arc : TTY=pts/3 ; PWD=/home/arc/gpmdp ; USER=root ; COMMAND=/usr/bin/pacman -U gpmdp-4.6.1-2-x86_64.pkg.tar.xz
dic 05 23:09:40 ARC-flex audit[17352]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=2 pid=17352 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:09:40 ARC-flex kernel: gpmdp[17352]: segfault at a4fa30 ip 0000000000a4fa30 sp 00007ffeeddc9788 error 15 in Google Play Music Desktop Player[200000+1602000]
dic 05 23:09:40 ARC-flex kernel: audit: type=1701 audit(1575608980.647:111): auid=1000 uid=1000 gid=1000 ses=2 pid=17352 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:09:41 ARC-flex systemd-coredump[17354]: Process 17352 (gpmdp) of user 1000 dumped core.
dic 05 23:09:46 ARC-flex audit[17373]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=2 pid=17373 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:09:46 ARC-flex kernel: gpmdp[17373]: segfault at a4fa30 ip 0000000000a4fa30 sp 00007fff892e5b48 error 15 in Google Play Music Desktop Player[200000+1602000]
dic 05 23:09:46 ARC-flex kernel: audit: type=1701 audit(1575608986.330:114): auid=1000 uid=1000 gid=1000 ses=2 pid=17373 comm="gpmdp" exe=2F7573722F73686172652F67706D64702F476F6F676C6520506C6179204D75736963204465736B746F7020506C61796572 sig=11 res=1
dic 05 23:09:46 ARC-flex systemd-coredump[17375]: Process 17373 (gpmdp) of user 1000 dumped core.

```

**Comment by lanceofwhichwedream** [comment]
Looks like this is fixed in version 4.7.1. Big props to @MarshallOfSound !!! Appreciate you and your hard work!
